lib/minitest.rb

```diff
@@ -655,6 +656,10 @@ def report
     def passed?
       true
     end
+
+    def mutex # :nodoc:
+      @mutex
+    end
```

Comment:
```suggestion
    def synchronize(&block) # :nodoc:
      @mutex.synchronize(&block)
    end
```
This would be cleaner interface-wise and avoid exposing the Mutex.
Also it would avoid needing changes in lib/minitest/parallel.rb

Reply:
👍  Thanks. It's better than my changes.